lib/xstate-ngx/src/lib/useActor.ts

```diff
@@ -26,7 +26,7 @@ import type { AnyMachineSnapshot } from 'xstate/dist/declarations/src/types';
  * @return The Angular service that provides the actor store.
  */
 export function useActor<TLogic extends AnyActorLogic>(
-  actorLogic: TLogic,
+  actorLogic: TLogic | (() => TLogic),
```

Comment:
I am not a fan of changing the public API, as this way it differs from the other xstate implementations. I am also not quite sure what you are trying to accomplish if it is purely to allow injection in the context that can already be accomplished with [lazy context](https://stately.ai/docs/context#lazy-initial-context) evaluation 
![image](https://github.com/user-attachments/assets/1a7dfba1-b373-45cb-b234-c302714a5830)

Reply:
I didn't realize that lazy-context could be useful here
but in the other hand, a Callback/Promise Actor? how do you inject there?

Reply:
you can pass input params between parent and child actors https://stately.ai/docs/invoke#input

Reply:
I still don't see a use case that is not already possible with existing APIs, can you share a code example?

Reply:
the QRActor, making it work without too much hassle,
consuming an Ionic Plugin, an API service, showing a Toast,
I don't see it easy without the Factory closure :thinking:

Reply:
https://github.com/niklas-wortmann/xstate-angular/blob/b5f3a33dc6f5f2f189598c7ffa611c1c91e4f627/apps/angular-xstate-demo/src/app/routing/routing.component.ts#L18

just like this? don't see how this is different

Reply:
That's true, that's just left over from my experiments. But I still don't see what this pattern isn't sufficient and just need an actorFactory?

Reply:
right, Callback and Promise actors are already factories, I was thinking about the State machines. I will continue my work and see if the current API is enough or not

Reply:
I realized this `runInInjectionContext` is in the Angular injectable.
I'm thinking on what can happen in an advanced scenario with actions triggered from other actors, events sent to the parent, etc. will these work ok?

Reply:
I'd assume so. If you happen to run into a case that doesn't work feel free to open a ticket with some sample code. In the meantime I will close this